httpstan/services_stub.py

```diff
@@ -116,11 +116,14 @@ async def call(
                 messages_file.write(message)
             # if `potential_readers == [socket_]` then either (1) no connections
             # have been opened or (2) all connections have been closed.
-            if not readable and potential_readers == [socket_]:
-                if future.done():  # type: ignore
-                    logger.debug(f"Stan services function `{function_basename}` returned or raised a C++ exception.")
+            if not readable:
+                if potential_readers == [socket_] and future.done():
+                    logger.debug(
+                        f"Stan services function `{function_basename}` returned without problems or raised a C++ exception."
```

Comment:
What other options there are?

- no problems
- C++ exception

Reply:
Those are the only options, I think.

Reply:
There might be a third option: the future could be canceled. This might happen if the subprocess is killed by some external program. Maybe an out of memory error would also kill the process and cancel the future.

If there's an easy way to artificially (sharply) limit the amount of memory available to a process, we could test this.